cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1872,6 +1872,105 @@ def get_distances(
             for _node in (self.leaves if leaves_only else self.nodes)
         }
 
+    def scale_to_unit_length(self) -> None:
+        """
+        Scales the tree to have unit length.
+
+        The longest path from root to leaf will have length 1 after the
+        scaling.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        current_times = self.get_times().values()
+        max_time = max(current_times)
+        min_time = min(current_times)
+        new_times = {}
+        for node in self.nodes:
+            new_times[node] = self.get_time(node) / (max_time - min_time)
+        self.set_times(new_times)
+
+    def get_depth(self) -> float:
+        """
+        Depth of the tree (in terms of time).
+
+        The depth of the tree (in terms of time) is defined as the greatest
+        time distance of any leaf of the tree from the root.
+
+        Returns:
+            Depth of the tree.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        times = self.get_times().values()
+        return max(times) - min(times)
+
+    def get_mutated_characters_along_edge(
+        self, parent: str, child: str
+    ) -> List[int]:
+        """
+        List of mutated characters along edge.
+
+        A character is considered to mutate if it goes from the zero state
+        to a non-zero state that is also not a missing state. In particular,
+        if a character goes missing, it is NOT considered as mutated by this
+        method.
+
+        Raises:
+            CassiopeiaTreeError if the edge does not exist or if the tree is
+                not initialized.
+        """
+        self.__check_network_initialized()
+
+        if child not in self.children(parent):
+            raise CassiopeiaTreeError("Edge does not exist.")
+
+        parent_states = self.get_character_states(parent)
+        child_states = self.get_character_states(child)
+        res = []
+        for i, (parent_state, child_state) in enumerate(
+            zip(parent_states, child_states)
+        ):
+            if (
+                parent_state == 0
+                and child_state != 0
+                and child_state != self.missing_state_indicator
+            ):
+                res.append(i)
+        return res
+
+    def get_unmutated_characters_along_edge(
+        self, parent: str, child: str
+    ) -> List[int]:
+        """
+        List of unmutated characters along edge.
+
+        A character is considered to not mutate if it goes from the zero state
+        to the zero state.
```

Comment:
Once again -- I wonder if we can be a bit more general.

I understand that we can't be as general as we'd like (returning all the characters that don't change along an edge) but what if we introduce an argument that _allows_ us to be that general? This would be along the lines of saying whether or not the mutation model is reversible or not.

Reply:
We could do this, but I don't think we need that behavior now (or ever). In fact I'm realizing that maybe we should also keep `get_mutated_characters_along_edge` as-is for the same reason. I'm just following [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) here.

Reply:
Haha, I hadn't heard of YAGNI before but I love it.

You're right -- let's not extend this function to the general case until we're sure we need it.